File: src/ObjectManager/Object.Tes/FilePacks/Records/040-LVSP.Leveled Spell.cs
using OA.Core;
using System.Collections.Generic;

namespace OA.Tes.FilePacks.Records
{
    public class LVSPRecord : Record
    {
        public override string ToString() => $"LVSP: {EDID.Value}";
        public STRVField EDID { get; set; } // Editor ID
        public BYTEField LVLD; // Chance
        public BYTEField LVLF; // Flags
        public List<LVLIRecord.LVLOField> LVLOs = new List<LVLIRecord.LVLOField>(); // Number of items in list

        public override bool CreateField(UnityBinaryReader r, GameFormatId format, string type, int dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "LVLD": LVLD = new BYTEField(r, dataSize); return true;
                case "LVLF": LVLF = new BYTEField(r, dataSize); return true;
                case "LVLO": LVLOs.Add(new LVLIRecord.LVLOField(r, dataSize)); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/040-LVSP.Leveled Spell.swift
//
//  LVSPRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class LVSPRecord: Record {
    public override var description: String { return "LVSP: \(EDID)" }
    public var EDID: STRVField = STRVField_empty // Editor ID
    public var LVLD: BYTEField! // Chance
    public var LVLF: BYTEField! // Flags
    public var LVLOs = [LVLIRecord.LVLOField]() // Number of items in list
    
    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID": EDID = r.readSTRV(dataSize)
        case "LVLD": LVLD = r.readT(dataSize)
        case "LVLF": LVLF = r.readT(dataSize)
        case "LVLO": LVLOs.append(LVLIRecord.LVLOField(r, dataSize))
        default: return false
        }
        return true
    }
}
